oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/web/server/logout/OidcClientInitiatedServerLogoutSuccessHandler.java

```diff
@@ -189,4 +182,90 @@ public void setLogoutSuccessUrl(URI logoutSuccessUrl) {
 		this.serverLogoutSuccessHandler.setLogoutSuccessUrl(logoutSuccessUrl);
 	}
 
+	/**
+	 * Set the {@link Converter} that converts {@link RedirectUriParameters} to redirect
+	 * URI
+	 * @param redirectUriResolver {@link Converter}
+	 * @since 6.4
+	 */
+	public void setRedirectUriResolver(Converter<RedirectUriParameters, Mono<String>> redirectUriResolver) {
+		Assert.notNull(redirectUriResolver, "redirectUriResolver cannot be null");
+		this.redirectUriResolver = redirectUriResolver;
+	}
+
+	/**
+	 * Parameters, required for redirect URI resolving.
+	 *
+	 * @author Max Batischev
+	 * @since 6.4
+	 */
+	public static class RedirectUriParameters {
+
+		private ServerWebExchange serverWebExchange;
```

Comment:
Please have these be `final` and in the constructor

Reply:
I can add Final to the class, but I'm not sure if it's worth making all the fields Final and initializing them in the constructor. The `DefaultRedirectUriResolver` requires two attributes `Authentication` and `ServerWebExchange` and this attributes are passed to the `onLogoutSuccess` method, meaning they cannot be null, so they can be made final and added to the constructor, `ClientRegistration` will be found in the resolver (`OidcClientInitiatedServerLogoutSuccessHandler. this.clientRegistrationRepository::findByRegistrationId`). That is, it cannot be declared final and required to be initialized in the constructor. I can easily pass it via setter if I need another one `ClientRegistration`:
```
this.handler.setRedirectUriResolver((params) -> {
			params.setClientRegistration(ClientRegistration.withRegistrationId("clientId")
					.build());
			return Mono.just("URL");});
```